src/app/tests/suites/commands/interaction_model/InteractionModel.h

```diff
@@ -143,40 +111,135 @@ class InteractionModelReports
+
+    InteractionModelReports & SetDataVersions(const std::vector<chip::DataVersion> & dataVersions)
+    {
+        mDataVersions.SetValue(dataVersions);
+        return *this;
+    }
+
+    InteractionModelReports & SetDataVersions(const chip::Optional<std::vector<chip::DataVersion>> & dataVersions)
+    {
+        mDataVersions = dataVersions;
+        return *this;
+    }
+
+    InteractionModelReports & SetIsUrgents(const std::vector<bool> isUrgents)
+    {
+        mIsUrgents.SetValue(isUrgents);
+        return *this;
+    }
+
+    InteractionModelReports & SetIsUrgents(const chip::Optional<std::vector<bool>> & isUrgents)
+    {
+        mIsUrgents = isUrgents;
+        return *this;
+    }
+
+    InteractionModelReports & SetEventNumber(const chip::Optional<chip::EventNumber> & eventNumber)
+    {
+        mEventNumber = eventNumber;
+        return *this;
+    }
+
+    InteractionModelReports & SetEventNumber(chip::EventNumber eventNumber)
+    {
+        mEventNumber.SetValue(eventNumber);
+        return *this;
+    }
+
+    InteractionModelReports & SetFabricFiltered(bool fabricFiltered)
+    {
+        mFabricFiltered.SetValue(fabricFiltered);
+        return *this;
+    }
+
+    InteractionModelReports & SetKeepSubscriptions(bool keepSubscriptions)
+    {
+        mKeepSubscriptions.SetValue(keepSubscriptions);
+        return *this;
+    }
+
+    InteractionModelReports & SetKeepSubscriptions(const chip::Optional<bool> & keepSubscriptions)
+    {
+        mKeepSubscriptions = keepSubscriptions;
+        return *this;
+    }
+
+    InteractionModelReports & SetAutoResubscribe(bool autoResubscribe)
+    {
+        mAutoResubscribe.SetValue(autoResubscribe);
+        return *this;
+    }
+
+    InteractionModelReports & SetAutoResubscribe(const chip::Optional<bool> & autoResubscribe)
+    {
+        mAutoResubscribe = autoResubscribe;
+        return *this;
+    }
+
+    InteractionModelReports & SetMinInterval(uint16_t minInterval)
+    {
+        mMinInterval = minInterval;
+        return *this;
+    }
+
+    InteractionModelReports & SetMaxInterval(uint16_t maxInterval)
+    {
+        mMaxInterval = maxInterval;
+        return *this;
+    }
+
+    void ResetOptions()
+    {
+        mDataVersions      = chip::NullOptional;
+        mIsUrgents         = chip::NullOptional;
+        mEventNumber       = chip::NullOptional;
+        mFabricFiltered    = chip::Optional<bool>(true);
+        mKeepSubscriptions = chip::NullOptional;
+        mAutoResubscribe   = chip::NullOptional;
+        mMinInterval       = 0;
+        mMaxInterval       = 0;
+    }
+
+    chip::Optional<std::vector<chip::DataVersion>> mDataVersions;
+    chip::Optional<std::vector<bool>> mIsUrgents;
+    chip::Optional<chip::EventNumber> mEventNumber;
+    chip::Optional<bool> mFabricFiltered;
+    chip::Optional<bool> mKeepSubscriptions;
+    chip::Optional<bool> mAutoResubscribe;
+    uint16_t mMinInterval;
+    uint16_t mMaxInterval;
```

Comment:
I'm really not convinced about storing this state in (static) command objects, which I think this is doing.

This state should be associated with a specific command execution, not with the command object itself.

Does something guarantee that two invocations of the command can't both end up touching these members, in interactive mode?